### PR TITLE
Runtime istio updated to 1.20.8 (ossm 2.6) and Istio go dep to 1.22.3

### PR DESCRIPTION
**Pull request by eguzki**
Golang istio.io/istio deps upgraded to 1.22.3

It is required because golang envoygateway 1.1 dep conflicts on github.com/envoyproxy/go-control-plane/envoy/extensions/injected_credentials/generic/v3 package

istio.io/istio 1.20.0 requires a package from github.com/envoyproxy/go-control-plane in 0.12.0 that does not exist when github.com/envoyproxy/go-control-plane is upgraded to 0.12.1 due to envoygateway 1.1

### CODE REVIEW

File: controllers/rate_limiting_envoygateway_envoyextensionpolicy_controller.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+//+kubebuilder:rbac:groups=gateway.envoyproxy.io,resources=envoyextensionpolicies,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=gateways,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=httproutes,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=kuadrant.io,resources=ratelimitpolicies,verbs=get;list;watch;update;patch
+
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *RateLimitingEnvoyExtensionPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	a := &egv1alpha1.EnvoyExtensionPolicy{}
```
Comment (eguzki) [inline controllers/rate_limiting_envoygateway_envoyextensionpolicy_controller.go:25]: This is just for adding a real package import, which affects how deps are organized in go.mod

**Comment by eguzki** [comment]
passing tests means that our operator having golang deps on istio 1.22.3 works well with the runtime version of istio being 1.20.8 (the expected one for OSSM 2.6)

File: go.mod [hunk 2]
```diff
@@ -1,72 +1,73 @@
 module github.com/kuadrant/kuadrant-operator
 
-go 1.22.0
+go 1.22.5
```
Comment (adam-cattermole) [inline go.mod:3]: Interesting that envoyproxy/gateway explicitly has .5 in their go.mod for v1.1

**Review by adam-cattermole — APPROVED** [review]
Looks good to me

**Comment by adam-cattermole** [comment]
I'm in need of these changes for the authpolicy branch (#737) now that the envoygateway branch has been rebased to gwapi v1.1. I hope you do not mind but I'm going to merge this to rebase and finish up the next draft of my PR.

**Pull request merged**